### PR TITLE
Feat/update swap data tracking

### PR DESCRIPTION
**Pull request by sarneijim**
### 📝 Description

Add data tracking to swap following the data tracking plan (linked resource)
Analytics and product has been testing this pr to check the necessities.

### ❓ Context

- **Impacted projects**: 
  ```LLD```
- **Linked resource(s)**: 
  - Jira https://ledgerhq.atlassian.net/browse/LIVE-3674
  - Data tracking definition https://ledgerhq.atlassian.net/wiki/spaces/PTX/pages/edit-v2/3733454849

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/components/BalanceInfos/index.jsx [hunk 1]
```diff
@@ -135,7 +136,7 @@ export default function BalanceInfos({ totalBalance, valueChange, isAvailable, u
   }, [history, ptxSmartRouting]);
 
   const onSwap = useCallback(() => {
-    setTrackingSource("Page Market");
+    setTrackingSource("Page Portafolio");
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-desktop/src/renderer/components/BalanceInfos/index.jsx:139]: Should be Portfolio

File: apps/ledger-live-desktop/src/renderer/components/BalanceInfos/index.jsx [hunk 2]
```diff
@@ -157,7 +158,17 @@ export default function BalanceInfos({ totalBalance, valueChange, isAvailable, u
...
         </Button>
 
-        <Button data-test-id="portfolio-swap-button" variant="color" onClick={onSwap}>
+        <Button
+          data-test-id="portfolio-swap-button"
+          variant="color"
+          event="button_clicked"
+          eventProperties={{
+            button: "swap",
+            page: "Page Portafolio",
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-desktop/src/renderer/components/BalanceInfos/index.jsx:167]: Portfolio

File: apps/ledger-live-desktop/src/renderer/families/cardano/MemoValueField.jsx [hunk 3]
```diff
@@ -25,13 +28,18 @@ const MemoValueField = ({
 
   const onMemoValueChange = useCallback(
     memo => {
+      track("button_clicked", {
```
Comment (juan-cortes) [inline apps/ledger-live-desktop/src/renderer/families/cardano/MemoValueField.jsx:31]: I assume product set these even names?
If not, perhaps something more meaningful would be better 😕

File: apps/ledger-live-desktop/src/renderer/screens/asset/index.jsx [hunk 4]
```diff
@@ -63,7 +63,7 @@ export default function AssetPage({ match }: Props) {
 
   return (
     <Box>
-      <TrackPage category="Potfolio" name="Asset allocation" currencyName={currency.name} />
+      <TrackPage category="Portfolio" name="Asset allocation" currencyName={currency.name} />
```
Comment (juan-cortes) [inline apps/ledger-live-desktop/src/renderer/screens/asset/index.jsx:66]: 🪴

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/index.jsx [hunk 5]
```diff
@@ -66,24 +67,65 @@ export default function ProviderRate({
...
+        return filter.every(item => [FILTER.decentralised, FILTER.float].includes(item));
+      });
+      if (defaultPartner) {
+        return defaultPartner;
+      } else if (decentralizedSwapAvailable && defaultDexPartner) {
+        return defaultDexPartner;
+      } else {
+        return {};
+      }
+    },
```
Comment (juan-cortes) [inline apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/index.jsx:85]: This seems unnecessarily complex. I'm sure you can simplify it, but perhaps I'm lacking context so I'll leave the approval to someone else. Same applies to [the logic inside the ProviderRate ](https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/Filter.jsx#L40-L69)components for that matter.

**Review by juan-cortes — COMMENTED** [review]
Just a few remarks, can't approve but perhaps it's ok

File: apps/ledger-live-desktop/src/renderer/families/cardano/MemoValueField.jsx (hunk 3 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/families/cardano/MemoValueField.jsx:31]: it is set by "analytics" team :neutral_face:

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/index.jsx (hunk 5 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/index.jsx]: You are right. Now the code is simpler :smiley:

**Review by juan-cortes — APPROVED** [review]
lgtm

**Review by ggilchrist-ledger — APPROVED** [review]
LGTM

**Pull request merged**